server/src/main/java/org/elasticsearch/search/lookup/LeafDocLookup.java

```diff
@@ -72,6 +79,10 @@ public void setDocument(int docId) {
 
     @Override
     public ScriptDocValues<?> get(Object key) {
+        // deprecate _type
+        if ("_type".equals(key)) {
+            DEPRECATION_LOGGER.deprecated(TYPES_DEPRECATION_MESSAGE);
```

Comment:
Should we use `deprecatedAndMaybeLog`?

Reply:
:) already done.  @rjernst mentioned the same thing.